[Bug]: Cannot find name 'PromiseWithResolvers'

### Describe the bug

A [reference](https://github.com/storybookjs/storybook/blob/9678d6f3774f3d4e0dcdc19bbf20735d670e7550/code/core/src/preview-api/modules/store/csf/portable-stories.ts#L265) to `PromiseWithResolvers` was introduced by https://github.com/storybookjs/storybook/pull/28383

That type is not imported anywhere in that file and my library (using react) doesn't compile because of it.

### Reproduction link

N/A

### Reproduction steps

_No response_

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.0/bin/npm <----- active
  Browsers:
    Chrome: 126.0.6478.127
    Firefox: 128.0
    Safari: 17.5
  npmPackages:
    @storybook/addon-a11y: ^8.2.1 => 8.2.1 
    @storybook/addon-actions: ^8.2.1 => 8.2.1 
    @storybook/addon-docs: ^8.2.1 => 8.2.1 
    @storybook/addon-essentials: ^8.2.1 => 8.2.1 
    @storybook/addon-interactions: ^8.2.1 => 8.2.1 
    @storybook/addon-links: ^8.2.1 => 8.2.1 
    @storybook/addon-webpack5-compiler-swc: ^1.0.4 => 1.0.4 
    @storybook/cli: ^8.2.1 => 8.2.1 
    @storybook/components: ^8.2.1 => 8.2.1 
    @storybook/manager-api: ^8.2.1 => 8.2.1 
    @storybook/node-logger: ^8.2.1 => 8.2.1 
    @storybook/react: ^8.2.1 => 8.2.1 
    @storybook/react-webpack5: ^8.2.1 => 8.2.1 
    @storybook/testing-library: ^0.2.2 => 0.2.2 
    @storybook/theming: ^8.2.1 => 8.2.1 
    storybook: ^8.1.11 => 8.2.1
```


### Additional context

_No response_

Hello I have the same issue on my side with the 8.2.1...
```
node_modules/@storybook/core/dist/preview-api/index.d.ts:349:22 - error TS2304: Cannot find name 'PromiseWithResolvers'.

349     renderingEnded?: PromiseWithResolvers<void>;
                         ~~~~~~~~~~~~~~~~~~~~

=> Failed to build the preview
SB_BUILDER-WEBPACK5_0003 (WebpackCompilationError): There were problems when compiling your code with Webpack.
Run Storybook with --debug-webpack for more information.

```

--- a/code/core/src/preview-api/modules/store/csf/portable-stories.ts
+++ b/code/core/src/preview-api/modules/store/csf/portable-stories.ts
@@ -263,10 +263,7 @@ type UnwrappedJSXStoryRef = {
   __pw_type: 'jsx';
   type: UnwrappedImportStoryRef;
 };
-type UnwrappedImportStoryRef = ComposedStoryFn & {
-  playPromise?: Promise<void>;
-  renderingEnded?: PromiseWithResolvers<void>;
-};
+type UnwrappedImportStoryRef = ComposedStoryFn;
 
 declare global {
   function __pwUnwrapObject(

